src/prefect/engine.py

```diff
@@ -2966,6 +2966,15 @@ async def create_autonomous_task_run(task: Task, parameters: Dict[str, Any]) ->
             factory = await ResultFactory.from_autonomous_task(task, client=client)
             await factory.store_parameters(parameters_id, parameters)
 
+        if not task.task_origin_hash:
```

Comment:
the only way we should get here is if someone tries to `submit` a `Task` they made via `Task(fn=obj)` instead of the standard decorator syntax

Reply:
Interesting. @zzstoatzz Are you saying that the example of a callable object doesn't work?

Reply:
it actually does!
> even though I seem to have no problem submitting runs of this task instance 🤔

but I'm saying without the `try/except` in `Task.__init__` that failsover by setting this hash to `None`, we fail some of the `test_tasks.py` tests where we are testing things about tasks instantiated in this way

I tried to reproduce the failure I saw in the test by creating a task from this callable class `A` and then submitting it, but it actually worked. So it wasn't the most satisfying resolution but wanted to flag in case someone had a better idea of what was happening